app/controllers/stories_controller.rb

```diff
@@ -28,6 +28,13 @@ def show
     render json: @story
   end
 
+  def sort
+    @stories = SortStories.new(params[:ordered_ids]).call do |story|
```

Comment:
It would be nice if you could apply authorize to whole @stories array here, so you'll don't need the block here.